cirq-web/cirq_ts/src/bloch_sphere/bloch_sphere.ts

```diff
@@ -0,0 +1,123 @@
+// Copyright 2021 The Cirq Developers
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import {Orientation} from './components/enums';
+import {Sphere} from './components/sphere';
+import {Axes} from './components/axes';
+import {Meridians} from './components/meridians';
+import {Labels} from './components/text';
+import {Vector} from './components/vector';
+
+import {BlochSphereScene} from './components/scene';
+import {Group, Scene, Vector3} from 'three';
+
+/**
+ * Class bringinging together the individual components like the
+ * Sphere, Axes, Meridians, and Text into the overall visualization
+ * of the Bloch sphere.
+ */
+export class BlochSphere extends Group {
+  private radius: number;
+  private hMeridians: number;
+  private vMeridians: number;
+
+  /**
+   * Class constructor.
+   * @param radius The radius of the Bloch sphere
+   * @param hMeridians The number of horizontal meridians desired for the Bloch sphere.
+   * @param vMeridians The number of vertical meridians desired for the Bloch sphere.
+   * @returns An instance of the class which can be easily added to a scene.
+   */
+  constructor(radius = 5, hMeridians = 7, vMeridians = 4) {
+    super();
+
+    if (radius < 1) {
+      throw new Error('Invalid radius provided in the BlochSphere constructor');
+    }
+
+    this.radius = radius;
+    this.hMeridians = hMeridians;
+    this.vMeridians = vMeridians;
+
+    this.addSphere();
+    this.addHorizontalMeridians();
+    this.addVerticalMeridians();
```

Comment:
Nit: this is missing the meridian in the XZ plane.

Reply:
For the sake of the visualization aesthetic I'd prefer to keep the meridians in the two planes..thoughts?

Reply:
This isn't a huge sticking point for me - I'm used to [QUIRK](https://algassert.com/quirk)'s Bloch sphere visualizations, which could be argued to have either two or three meridians. OK to leave as-is.